hub.py

```diff
@@ -261,20 +261,51 @@ def setup_nfs_share(self):
         ])
 
 
-    def deploy(self, auth_provider, proxy_secret_key):
+    def apply_hub_template_fixes(self, generated_config, proxy_secret_key):
         """
-        Deploy this hub
+        Modify generated_config based on what hub template we're using.
+
+        Different hub templates require different pre-set config. For example,
+        anything deriving from 'base-hub' needs all config to be under a 'base-hub'
+        config. dask hubs require apiTokens, etc.
+
+        Ideally, these would be done declaratively. Untile then, let's put all of
+        them in this function.
         """
         hub_template = self.spec['template']
 
-        generated_values = self.get_generated_config(auth_provider, proxy_secret_key)
         # FIXME: Have a templates config somewhere? Maybe in Chart.yaml
         # FIXME: This is a hack. Fix it.
         if hub_template != 'base-hub':
-            generated_values = {
-                'base-hub': generated_values
+            generated_config = {
+                'base-hub': generated_config
+            }
+
+        # LOLSOB FIXME
```

Comment:
Does #LOLSOB mean :smile:  + :sob:? hehe
Should we open an issue about fixing this later to not block the PR?